### DISCUSSION TITLE
CLI: supabase link returns: 'failed SASL auth (invalid SCRAM server-final-message received from server)'

### DISCUSSION THREAD

**Discussion by bobiboMC** [supabase/supabase#21509] [Questions] [answered]
Hi,
I'm using windows 11 and Visual Code.
I'm trying to use `yarn supabase link --project-ref <proj_ref>` but I always get this error in terminal:
![problem_auth](https://github.com/supabase/supabase/assets/61916959/4ecc32d6-cb20-4ef9-ad3a-55114fe145d9)

The steps i'm doing until I receive the error
1. I set my SUPABASE_DB_PASSWORD using:
   `$env:SUPABASE_DB_PASSWORD ='<password>'`
   and verify using `$env:SUPABASE_DB_PASSWORD`  that I indeed set it.
2. After that I run `yarn supabase login`
3. I run `yarn supabase link --project-ref <proj_ref>`
 And then I get this error.
 My SUPABASE_DB_PASSWORD has upper case characters, lower case characters, numbers and two period characters.
 I would glad for help.

**Comment by GaryAustin1** [comment]
![image](https://github.com/supabase/supabase/assets/54564956/9a2713c8-65c3-4d02-8598-416d11f5f3b8)
https://github.com/orgs/supabase/discussions/17817

**Reply by bobiboMC** [reply]
Hi, I'm already using the latest version `1.145.4`:
![image](https://github.com/supabase/supabase/assets/61916959/4bbe453c-1b85-4294-8eac-f125a4ce37ab)

**Reply by GaryAustin1** [reply, marked as answer]
I'm not up to speed on the CLI and passwords...
https://supabase.com/docs/guides/database/postgres/roles#special-symbols-in-passwords
You might try without the periods in your password.
Also note after 3 failed attempts your ip may be locked for 30 minutes.
